### DISCUSSION TITLE
How do I create a quarto filter which can update the description of a post listing?

### DISCUSSION THREAD

**Discussion by mckabue** [quarto-dev/quarto-cli#9767] [Q&A] [answered]
### Description

From my tests ([and from code](https://github.com/quarto-dev/quarto-cli/blob/e1026cf2cc1c91febbf4631b38f91ef7848f4f33/src/project/types/website/listing/website-listing-read.ts#L1110C5-L1112C70)), it appears that the message of a listing post cannot be updated on runtime (when quarto is rendering the HTML).

A want a lua filter that overrides this functionality and instead generates a custom description will be used by the listing page.

**Comment by mcanouil** [comment]
Could you share what you tried concretely for your filter?

**Reply by mckabue** [reply]
posts/welcome/index.qmd
```
---
title: "Welcome To My Blog"
author: "Tristan O'Malley"
date: "2024-05-01"
categories: [news]
filters:
  - at: pre-ast
    path: description-update.lua
---

This is the first post in a Quarto blog. Welcome!

![](thumbnail.jpg)

Since this post doesn't specify an explicit `image`, the first image in the post will be used in the listing page of posts.
```


posts/welcome/description-update.lua
```
return {
  ---@param doc pandoc.Pandoc
  ---@return pandoc.Pandoc
  Pandoc = function (doc)
    local meta = doc.meta
    meta['description'] = pandoc.Str('This is an updated description')
    
    return pandoc.Pandoc(doc.blocks, meta)
  end
}
```


Display In the post page:
<img width="758" alt="Screenshot 2024-05-26 at 06 06 06" src="https://github.com/quarto-dev/quarto-cli/assets/6072072/34b6d949-0956-4050-9dd2-92129252bbfe">

Display In the default Listings Page:
<img width="758" alt="Screenshot 2024-05-26 at 06 05 48" src="https://github.com/quarto-dev/quarto-cli/assets/6072072/2f27ed36-4b9b-4266-8be8-a1fb82d56e91">

**Reply by cscheid** [reply, marked as answer]
I don't think that's possible today, unfortunately. I've opened an issue to track: #9811
